[MRG] MNT: Drop `n_tasks` in `enet_coordinate_descent`

jakirkham opened the pull request:
In `enet_coordinate_descent` and `sparse_enet_coordinate_descent`, `y` is always C-contiguous as it is required by the function itself. So there is no need to compute the stride length of `y`, `n_tasks`. Instead it can be treated as `1`, which it is.

jakirkham committed:
MNT: Drop `n_tasks` in `enet_coordinate_descent`

In `enet_coordinate_descent`, `y` is always C-contiguous as it is
required by the function itself. So there is no need to compute the
stride length of `y`, `n_tasks`. Instead it can be treated as `1`, which
it is.

agramfort commented:
should this also be cleaned up in 

https://github.com/jakirkham/scikit-learn/blob/170028f38b2686a2d08cba94abc143e02e437050/sklearn/linear_model/cd_fast.pyx#L336

?

jakirkham committed:
MNT: Drop `n_tasks` in `enet_coordinate_descent`

In `sparse_enet_coordinate_descent`, `y` is always C-contiguous as it is
required by the function itself. So there is no need to compute the
stride length of `y`, `n_tasks`. Instead it can be treated as `1`, which
it is.

jakirkham commented:
Sure, done.

agramfort commented:
thx @jakirkham
